apps/frontend/src/lib/langgraph/client.ts

```diff
@@ -25,11 +26,11 @@ export async function getOrCreateThread(client: Client): Promise<Thread<DefaultV
 	if (existingThreads.length > 0) {
 		const existingThread = existingThreads[0];
 		console.info('Using existing thread', existingThread);
-		return existingThread;
+		return existingThread as Thread<ThreadValues>;
 	} else {
 		console.info('No existing thread found, creating anew');
 		const thread = await client.threads.create();
-		return thread;
+		return thread as Thread<ThreadValues>;
```

Comment:
Gold standard here is explicit type guards/assertions (out of scope for now), just so you know. This ensures that there's a hard error whenever an assumption about data structures is invalid.

Normally, the API layer should take care of validating types using a proper validation library (e.g. using JSONSchema). So LangGraph's client API should do that! ;)

Just mentioning it here for reference/educational purposes. https://www.robinwieruch.de/typescript-type-guard/